gio/src/subclass/loadable_icon.rs

```diff
@@ -0,0 +1,134 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use super::icon::IconImpl;
+use crate::{AsyncResult, Cancellable, InputStream, LoadableIcon};
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, GString, IsA};
+
+#[derive(Debug)]
+pub struct AsyncReadyCallback {
+    pub callback: ffi::GAsyncReadyCallback,
+    pub user_data: glib::ffi::gpointer,
+}
+
+pub trait LoadableIconImpl: IconImpl {
+    fn load(
+        &self,
+        icon: &Self::Type,
+        size: i32,
+        cancellable: Option<&Cancellable>,
+    ) -> Result<(InputStream, Option<GString>), glib::Error>;
+    fn load_async(
+        &self,
+        icon: &Self::Type,
+        size: i32,
+        cancellable: Option<&Cancellable>,
+        callback: AsyncReadyCallback,
```

Comment:
Maybe this can all be done a bit nicer with the new `GTask` bindings? I don't think it's actually usable the way you currently do it here